templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -22,10 +22,20 @@ contents:
     # If $DHCP6_FQDN_FQDN is not empty and is not localhost.localdomain
     [[ -n "$DHCP6_FQDN_FQDN" && "$DHCP6_FQDN_FQDN" != "localhost.localdomain" && "$DHCP6_FQDN_FQDN" =~ "." ]] && hostnamectl set-hostname --static --transient $DHCP6_FQDN_FQDN
     case "$STATUS" in
-        up|down|dhcp4-change|dhcp6-change)
-        logger -s "NM resolv-prepender triggered by ${1} ${2}."
+        up|dhcp4-change|dhcp6-change)
+        >&2 echo "NM resolv-prepender triggered by ${1} ${2}."
 
         # Ensure resolv.conf exists before we try to run podman
+        # In DHCP connections, the resolv.conf content may be late, thus we wait for nameservers
+        timeout 45s /bin/bash <<EOF
+            if [[ "$STATUS" == dhcp* ]]; then
+                >&2 echo  "NM resolv-prepender: Checking for nameservers" /var/run/NetworkManager/resolv.conf
```

Comment:
Nit: This line is formatted/worded a bit oddly. Might be better as:

\>&2 echo  "NM resolv-prepender: Checking for nameservers in /var/run/NetworkManager/resolv.conf"